### ISSUE TITLE
define a WebUI API that userscripts can make use of

### ISSUE THREAD

**Issue by SoniEx2** [tootsuite/mastodon#13947]
### Pitch

the WebUI should have an API for userscripts to make use of.

### Motivation

it's currently impossible to integrate JS with the WebUI. userstyles work fine, but there's only so much they can do. adding an API for userscripts to use would solve that.

**Comment by YourAutisticLife** [comment]
Yep. I wanted to try implementing some experimental timeline cleanup algorithms but I cannot right now due to the lack of an API. It stings very badly that this was reported three years ago and nothing has happened.
